cmd/influx/cli/cli.go

```diff
@@ -139,56 +154,57 @@ func (c *CommandLine) Run() {
 
 	c.Version()
 
-	var historyFile string
+	var historyFilePath string
 	usr, err := user.Current()
-	// Only load history if we can get the user
+	// Only load/write history if we can get the user
 	if err == nil {
-		historyFile = filepath.Join(usr.HomeDir, ".influx_history")
-
-		if f, err := os.Open(historyFile); err == nil {
-			c.Line.ReadHistory(f)
-			f.Close()
+		historyFilePath = filepath.Join(usr.HomeDir, ".influx_history")
+		if c.historyFile, err = os.OpenFile(historyFilePath, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0640); err == nil {
+			defer c.historyFile.Close()
+			c.Line.ReadHistory(c.historyFile)
 		}
 	}
 
+	// read from prompt until exit is run
 	for {
-		l, e := c.Line.Prompt("> ")
-		if e != nil {
-			break
-		}
-		if c.ParseCommand(l) {
-			// write out the history
-			if len(historyFile) > 0 {
+		select {
+		case <-c.osSignals:
+			close(c.Quit)
+		case <-c.Quit:
+			// write to history file
+			_, err := c.Line.WriteHistory(c.historyFile)
+			if err != nil {
+				fmt.Printf("There was an error writing history file: %s\n", err)
+			}
+			// exit CLI
+			os.Exit(0)
```

Comment:
I think we really need a "close" function here.  Then we can do something like:

```go
if c.Line != nil {
    c.Line.Close()
    c.Line = nil
}
```

And any other cleanup we need.

If we do an `os.Exit` without closing the liner down that is what will leave your terminal in a bad state.